azure/scope/managedcontrolplane.go

```diff
@@ -273,6 +273,15 @@ func (s *ManagedControlPlaneScope) GroupSpecs() []azure.ASOResourceSpecGetter[*a
 			Owner:          *metav1.NewControllerRef(s.ControlPlane, infrav1.GroupVersion.WithKind(infrav1.AzureManagedControlPlaneKind)),
```

Comment:
I skipped out on adding the unit tests from `GroupSpecs()` here and in `cluster.go` due to the `owner` still being set in this version, and we can't test the owner being set correctly since it sets some fields to the address of a bool.